### PR TITLE
Add endpoint for getting a role's users

### PR DESCRIPTION
**Pull request by robines**
Role is basically just a group of permissions. It's not directly tied to users. To tie a role to a user, one must use either UserOrgRole, UserGangRole, or UserGangSectionRole. This will give the role to the user on a specific hierarchical level (on a specific object instance of an Organization, Gang or Section). 

This PR adds an endpoint which gets all UserOrgRole/UserGangRole/UserGangSectionRoles and their users, based on the specified role id. The object is also fetched, meaning we can easily it in frontend.

Response looks something like:

```jsonc
[
  {
    "user": UserDto,
    // org_role when role is given on Org level, gang_role for Gang, and section_role for Section
    "gang_role": {
      "created_at": "2024-10-29T23:37:39.950095Z",
      "created_by": UserDto, // Who gave the role
      // organization when org_role, gang when gang_role, section when section_role
      "gang": {
        "id": 4,
        "name_nb": "Markedsføringsgjengen",
        "name_en": "Markedsføringsgjengen",
        "abbreviation": "MG",
        ...
      }
    }
  },
  ...
]
```

### CODE REVIEW

**Review by Snorre98 — DISMISSED** [review]
LGTM.

**Review by Snorre98 — APPROVED** [review]
LGTM

e9c-232712975

File: backend/samfundet/views.py [hunk 1]
```diff
@@ -324,6 +328,22 @@ class RoleView(ModelViewSet):
...
+    def users(self, request: Request, pk: int) -> Response:
+        role = get_object_or_404(Role, id=pk)
+
+        org_roles = UserOrgRole.objects.filter(role=role).select_related('user', 'obj')
+        gang_roles = UserGangRole.objects.filter(role=role).select_related('user', 'obj')
+        section_roles = UserGangSectionRole.objects.filter(role=role).select_related('user', 'obj')
+
+        org_data = UserOrgRoleSerializer(org_roles, many=True).data
+        gang_data = UserGangRoleSerializer(gang_roles, many=True).data
+        section_data = UserGangSectionRoleSerializer(section_roles, many=True).data
```
Comment (emilte) [inline backend/samfundet/views.py:341]: Hvorfor ønsker dere å sende dette til frontend?

Comment (robines) [reply backend/samfundet/views.py:341]: @emilte For å kunne se hvem som har en gitt rolle, pluss info om hvem som ga den, og hvilken org/gjeng/seksjon den ble satt til

**Pull request merged**